### PR TITLE
1/init

### PR DESCRIPTION
**Pull request by Julien-Torrent**

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const { hiddenTagId } = options;
+
+  const iTS = new ItemTagService();
+  const taskManager = new ItemTagTaskManager(iS, iMS, iTS, itemTaskManager);
+
+  const isItemHidden = async (item: Item, actor: Actor, log: FastifyLoggerInstance) => {
+    const t1 = taskManager.createGetOfItemTask(actor as Member, item);
+    const t2 = membershipTaskManager.createGetMemberItemMembershipTask(actor, {
+      item,
+      validatePermission: 'admin' as PermissionLevel,
```
Comment (pyphilia) [inline src/plugin.ts:27]: I doesn't work if used with PermissionLevel.Admin?

File: src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    });
+    t2.getInput = () => {
+      t2.skip = t1.result.filter(({ tagId }) => tagId === hiddenTagId).length <= 0;
```
Comment (pyphilia) [inline src/plugin.ts]: Maybe it's clearer if you use `!Boolean(...length)`

File: src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    items: { dbService: iS, taskManager: itemTaskManager },
+    itemMemberships: { dbService: iMS, taskManager: membershipTaskManager },
+    taskRunner: runner,
+  } = fastify;
+
+  const { hiddenTagId } = options;
+
+  const iTS = new ItemTagService();
+  const taskManager = new ItemTagTaskManager(iS, iMS, iTS, itemTaskManager);
+
```
Comment (pyphilia) [inline src/plugin.ts:21]: Add a comment that it is hidden only if the user does not have admin permission over it.

File: src/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    async (items, actor, { log }) => {
+      const filteredItems = await Promise.all(
+        items.map(async (item) => {
+          try {
+            await isItemHidden(item, actor, log);
+            return item;
+          } catch (err) {
+            if (isGraaspError(err)) {
+              return null as unknown as Item<UnknownExtra>;
+            }
```
Comment (pyphilia) [inline src/plugin.ts]: It doesn't look good (the typings). I think it needs some comments as well.

File: src/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            throw err;
+          }
+        }),
+      );
+      items.splice(0, items.length, ...filteredItems.filter(Boolean));
```
Comment (pyphilia) [inline src/plugin.ts:83]: I guess it was to replace in-place the items? Add comment.

File: src/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    async (items, actor, { log }) => {
+      const filteredItems = await Promise.all(
+        items.map(async (item) => {
+          try {
+            await isItemHidden(item, actor, log);
+            return item;
+          } catch (err) {
+            if (isGraaspError(err)) {
+              return null as unknown as Item<UnknownExtra>;
+            }
```
Comment (pyphilia) [inline src/plugin.ts]: Same as above.

File: src/plugin.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            throw err;
+          }
+        }),
+      );
+      items.splice(0, items.length, ...filteredItems.filter(Boolean));
```
Comment (pyphilia) [inline src/plugin.ts:105]: Same as above.

File: tsconfig.json [hunk 8]
```diff
@@ -22,10 +22,9 @@
     // "reactNamespace": "",                             /* Specify the object invoked for `createElement`. This only applies when targeting `react` JSX emit. */
     // "noLib": true,                                    /* Disable including any library files, including the default lib.d.ts. */
     // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
-
     /* Modules */
     "module": "commonjs" /* Specify what module code is generated. */,
-    // "rootDir": "./",                                  /* Specify the root folder within your source files. */
+    // "rootDir": "./src", /* Specify the root folder within your source files. */
```
Comment (pyphilia) [inline tsconfig.json]: Indent like the other lines?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice 🍓 
Do you think you could already do the public plugin similarly to the ones I've done?

**Comment by Julien-Torrent** [comment]
I probably won't have time to do it

File: src/plugin.ts (hunk 1 above)
Comment (Julien-Torrent) [reply src/plugin.ts:27]: No, if PermissionLevel.Admin, when running the plugin cannot find the graasp module

File: src/plugin.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+          } catch (err) {
+            if (isGraaspError(err)) {
+              return null;
+            }
+            throw err;
+          }
+        }),
+      );
+
+      // Remve all hidden items in-place because we can't return a value from the handler
```
Comment (pyphilia) [inline src/plugin.ts]: `Remove`

**Review by pyphilia — APPROVED** [review]

**Pull request merged**